### PR TITLE
CE-655

### PR DESCRIPTION
**Pull request by rogamoore**
- specify version for FOSUserBundle
- activate FOSUserBundle reset functions
- disable unnecessary routes
- override FOSUserBundle templates
- separate User and Profile Controller and update permissions
- create UserDateTimeListener to update session in a single place
- add Listener to redirect after reset to homepage

### CODE REVIEW

File: Controller/UserController.php [hunk 1]
```diff
@@ -11,41 +11,110 @@
...
+    {
         $userManager = $this->get('fos_user.user_manager');
         $users = $userManager->findUsers();
 
         return $this->render('CampaignChainCoreBundle:User:index.html.twig',
             array(
-                'users' =>   $users,
-                'page_title' => 'Users',
+                'users' => $users,
+                'page_title' => 'Current Users',
```
Comment (groganz) [inline Controller/UserController.php]: Why did you add "Current" to the title?

File: Resources/views/Resetting/email.txt.twig [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{% trans_default_domain 'FOSUserBundle' %}
+{% block subject %}
+Reset Password - CampaignChain
```
Comment (groganz) [inline Resources/views/Resetting/email.txt.twig]: The name of the application (in this case "CampaignChain") should be filled in dynamically from the [name property of the System entity](https://github.com/CampaignChain/core/blob/master/Entity/System.php#L44). Reason being that we want to allow this to be configurable at a later point, so that a user running their own in-house instance could name it "ACME Campaign Manager", for example.

File: Resources/views/Resetting/email.txt.twig [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+{% endblock %}
+{% block body_text %}
+Hello {{ user.username }},
+
+To reset your password - please visit
+{{ confirmationUrl }}
+
+Regards,
+The CampaignChain Team
```
Comment (groganz) [inline Resources/views/Resetting/email.txt.twig]: Again, replace "CampaignChain" with the System name, see above.

File: Resources/views/Resetting/email.txt.twig (hunk 2 above)
Comment (rogamoore) [inline Resources/views/Resetting/email.txt.twig]: I agree. However in a future version it would be better to make all e-mail templates configurable and translatable.
